### ISSUE TITLE
Unable to provision GKE cluster from Rancher UI on upgraded upstream cluster

### ISSUE THREAD

**Issue by thehejik** [rancher/rancher#32124]
**What kind of request is this (question/bug/enhancement/feature request):** bug

**Steps to reproduce (least amount of steps as possible):**
* **HAS BEEN ADDED 04/16**: Deploy rancher:2.5.7 on upstream cluster first and perform upgrade to `v2.5-head`
* In Rancher UI v2.5-head `4e70ed2` create a cloud credentials for google account (import json for used service account)
* Click `Add Cluster` in Rancher/Ember UI, select hosted `Google GKE`
* Select the credentials, enter project id (rancher-qa)
* Keep default values for cloud but change used network to "default"
* Click `Save` button

**Result:**
* The `Save` button is changed to `Saving ...` and will stay like this forever
* Uncaught exceptions are visible in Web developer in Console tab:
```
  TypeError: Cannot read property 'googleCredentialSecret' of undefined
  TypeError: Cannot read property 'network' of undefined
  TypeError: Cannot read property 'subnetwork' of undefined
  TypeError: Cannot read property 'privateClusterConfig' of undefined
  Error: Property set failed: object in path "config.ipAllocationPolicy" could not be found
```
[full webdeveloper_console.log](https://github.com/rancher/rancher/files/6309848/webdeveloper_console.log)

![gke_provisioning_issue](https://user-images.githubusercontent.com/12828077/114679679-ae0a2180-9d0c-11eb-9701-9e44c8098781.png)
* No traces of any resources in Google Cloud Console

**Other details that may be helpful:**
* pod `gke-config-operator` using image `rancher/gke-operator:v1.0.1-rc6` (no logs present)
* I was able provision a GKE cluster on rancher:2.5.6 by using the same service account (json) and project id

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.5-head `4e70ed2`
- Installation option (single install/HA): HA

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): GKE Hosted
- Machine type (cloud/VM/metal) and specifications (CPU/memory): default (n1-standart-2)

**Comment by prabalsharma** [comment]
I was not able to reproduce this on v2.5-4e70ed29ccc5de7c61516ba0367c7c4904989b56-head
I used default network and subnet.

**Comment by cbron** [comment]
@westlywright this may already be fixed ^

**Comment by westlywright** [comment]
@cbron @gaktive I could not repro either. Can we close this?

**Comment by cbron** [comment]
Sure, @thehejik / @prabalsharma reopen if you see again.

**Comment by sowmyav27** [comment]
Reopening this as I see this issue

**Comment by thehejik** [comment]
> **Note:**
> 
> * reproducing it consistently with the default values set in the UI on a HA setup and _not_ on a single node install on the same commit of 2.5-head

Good point, it is working correctly on upstream docker/k3s/singlenode as well as on RKE/singlenode|HA when `rancher:v2.5-head` `41f173a` is deployed directly. 

It doesn't work for me on any upstream RKE clusters where `rancher:2.5.7` was deployed first (uses GKEv1) and then upgraded to `rancher:v2.5-head` over `helm upgrade ...`.

So correct **reproducer** for this issue is to deploy `rancher:2.5.7` first and then upgrade to `rancher:v2.5-head`.

**Comment by westlywright** [comment]
Not a UI issue. The UI sends the data to the API, the API accepts the call but almost immediately the `gkeConfig` is removed from the API object which sends an update to the UI. This happens so fast the UI can't transition from the route and the components on the page are looking for items in the `gkeConfig` which no longer exists. I have updated @cmurphy and @sowmyav27.

**Comment by cmurphy** [comment]
The problem is caused by rancher-operator. The fix is in rancher-operator v0.1.4-rc3, but it looks like Rancher doesn't automatically upgrade the rancher-operator Deployment when it upgrades itself.

**Comment by sowmyav27** [comment]
**On an upgrade from 2.5.7 to 2.5-head commit id: `124e7af`**

Able to deploy a GKEv2 cluster on an upgraded rancher server setup. Cluster comes up Active.
